<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Methods in population genomics | Yann X.C. Bourgeois </title> <meta name="author" content="Yann X.C. Bourgeois"> <meta name="description" content="Ongoing project"> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" href="/assets/css/scholar-icons.css?62b2ac103a88034e6882a5be5f3e2772"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="/assets/img/palmtreeicon.png?17bc196ea3bf2641d4cdcb8d2cfe16d6"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://yannbourgeois.github.io/projects/4_project/"> <script src="/assets/js/theme.js?9a0c749ec5240d9cda97bc72359a72c0"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>
    initTheme();
  </script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> <span class="font-weight-bold">Yann</span> X.C. Bourgeois </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item active"> <a class="nav-link" href="/projects/">projects <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">repositories </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="nav-item "> <a class="nav-link" href="/teaching/">teaching </a> </li> <li class="nav-item "> <a class="nav-link" href="/people/">people </a> </li> <li class="nav-item dropdown "> <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">submenus </a> <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown"> <a class="dropdown-item " href="/publications/">publications</a> <div class="dropdown-divider"></div> <a class="dropdown-item " href="/projects/">projects</a> <div class="dropdown-divider"></div> <a class="dropdown-item " href="/blog/">blog</a> </div> </li> <li class="nav-item"> <button id="search-toggle" title="Search" onclick="openSearchModal()"> <span class="nav-link">ctrl k <i class="ti ti-search"></i></span> </button> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Methods in population genomics</h1> <p class="post-description">Ongoing project</p> </header> <article> <p>There are many methods in population genomics. To help newcomers in the field, we published a review on the main methods in the field, available <a href="https://doi.org/10.1111/mec.15989" rel="external nofollow noopener" target="_blank">here</a></p> <table> <thead> <tr> <th>Software</th> <th>Class of method</th> <th>Purpose</th> <th>Specifics</th> <th>Issues and warnings</th> <th>Link</th> <th>Reference</th> </tr> </thead> <tbody> <tr> <td>POPGenome</td> <td>Differentiation/Diversity/Recombination</td> <td>Computing summary statistics based on AFS and LD along genomes</td> <td>Accepts VCF and GFF/GFT files, efficient and fast. Tests for admixture available (ABBA BABA test). Includes basic coalescence simulations (ms and msms)</td> <td>Mostly limited to summary statistics (but coalescent simulations are possible). No built-in SNP calling module</td> <td>http://catchenlab.life.illinois.edu/stacks/</td> <td>(Pfeifer et al., 2014)</td> </tr> <tr> <td>ANGSD</td> <td>Differentiation/Diversity/Recombination</td> <td>Computing summary statistics based on AFS and LD along genomes</td> <td>Able to process BAM files, built-in procedures for data filtering, admixture analysis. Suited for low-depth data. Includes a suite of methods to estimate relatedness (NGSRelate).</td> <td>Mostly limited to summary statistics. Tutorials not always up-to-date.</td> <td>https://github.com/ANGSD/angsd ; https://github.com/ANGSD/NgsRelate</td> <td>(Korneliussen et al., 2014; Hanghøj et al., 2019)</td> </tr> <tr> <td>Arlequin</td> <td>Differentiation/Diversity/Recombination</td> <td>Computing summary statistics based on AFS and LD along genomes</td> <td>Can output AFS for further analysis in fastsimcoal2</td> <td>Slower than PopGenome, requires a specific format and file conversion.</td> <td>http://cmpg.unibe.ch/software/arlequin35/Arl35Downloads.html</td> <td>(Excoffier and Lischer, 2010)</td> </tr> <tr> <td>VCFTOOLS</td> <td>Differentiation/Diversity/Recombination</td> <td>Computing summary statistics based on AFS and LD along genomes</td> <td>Fast. VCFTOOLS can also be used for SNP filtering</td> <td>Less summary statistics than POPGenome</td> <td>https://vcftools.github.io/man_latest.html</td> <td>(Danecek et al., 2011)</td> </tr> <tr> <td>ATLAS</td> <td>Differentiation/Diversity/Recombination</td> <td>Low depth sequencing/ancient samples analsis</td> <td>Particularly suited for analyzing ancient samples. Includes sets of tools to call variants, estimate post-mortem damage, inbreeding, genetic diversity. Produces the input file for PSMC (demography from a single diploid genome)</td> <td>Better used in combination with GATK pipelines. Still in development</td> <td>https://bitbucket.org/wegmannlab/atlas/wiki/Home</td> <td>(Link et al., 2017)</td> </tr> <tr> <td>POPTREE2</td> <td>Genetic differentiation</td> <td>Visualizing a matrix of pairwise differentiation statistics as a tree</td> <td>Can be used for pooled datasets, several statistics can be used</td> <td>Differentiation measures alone do not necessarily retrieve the actual history of populations</td> <td>http://www.med.kagawa-u.ac.jp/~genomelb/takezaki/poptree2/index.html</td> <td>(Takezaki et al., 2010)</td> </tr> <tr> <td>EEMS</td> <td>Landscape genomics</td> <td>Estimating barriers to gene flow in a spatial context</td> <td>Estimates pairwise relatedness between all samples, and compares it to isolation-by-distance expectations to identify barriers to gene flow and corridors of higher connectivity. Can handle both haploid and diploid data</td> <td>Requires to convert VCF file into PLINK binary format. Estimates effective migration rates (does not disantangle migration rates and effective population sizes). Setting parameters for the MCMC chain requires some trial-and-error</td> <td>https://github.com/dipetkov/eems</td> <td>(Petkova et al., 2015)</td> </tr> <tr> <td>MAPS</td> <td>Landscape genomics</td> <td>Estimating barriers to gene flow in a spatio-temporal context</td> <td>Expands on EEMS, but takes into account the phase to reconstruct past changes in connectivity. Can disentantle migration rates and effective population sizes (unlike EEMS)</td> <td>Relies on identity-by-descent tracks, requiring phasing (for example using BEAGLE). A pipeline to obtain IBD tracks is available, with a few details here: https://github.com/halasadi/ibd_data_pipeline/issues/1</td> <td>https://github.com/halasadi/MAPS</td> <td>(Al-Asadi et al., 2019)</td> </tr> <tr> <td>TESS3R</td> <td>Landscape genomics</td> <td>Grouping individuals in clusters maximizing HW equilibrium and LD between loci</td> <td>Incorporates geographic information of samples. Can run genome scans of selection based on contrasting ancestral and modern allele frequencies.</td> <td>Importing data requires using conversion tools found in the LEA suite</td> <td>https://bcm-uga.github.io/TESS3_encho_sen/</td> <td>(Caye et al., 2016)</td> </tr> <tr> <td>SNPRelate</td> <td>Multivariate analysis</td> <td>Summarizing variance across loci and visualizing inter-individual genetic distance</td> <td>Fast. Can use VCF files as an input</td> <td>Requires careful interpretation (Jombard et al. 2009)</td> <td>https://bioconductor.org/packages/release/bioc/html/SNPRelate.html</td> <td>(Zheng et al., 2012)</td> </tr> <tr> <td>Eigenstrat/smartpca</td> <td>Multivariate analysis</td> <td>Summarizing variance across loci and visualizing inter-individual genetic distance</td> <td>Fast. Can use VCF files as an input</td> <td>Requires careful interpretation (Jombard et al. 2009)</td> <td>https://github.com/DReichLab/EIG/tree/master/EIGENSTRAT</td> <td>(Price et al., 2006)</td> </tr> <tr> <td>DAPC (adegenet)</td> <td>Multivariate analysis/Clustering</td> <td>Maximizes divergence between groups identified by PCA</td> <td>Fast. Less sensitive to HWE assumptions. Claims to be more efficient than Structure</td> <td>Requires careful interpretation (Jombard et al. 2009)</td> <td>http://adegenet.r-forge.r-project.org/</td> <td>(Jombart et al., 2010)</td> </tr> <tr> <td>sPCA (adegenet)</td> <td>Multivariate analysis/Clustering</td> <td>Spatially explicit model to assess population structure</td> <td>Spatially explicit and able to detect cryptic structure. Fast.</td> <td>Does not take into account HW equilibrium or LD</td> <td>http://adegenet.r-forge.r-project.org/</td> <td>(Jombart et al., 2008)</td> </tr> <tr> <td>LAMP</td> <td>Pedigree, Identity by descent/state</td> <td>Chromosome painting, relatedness</td> <td>LAMP also allows for association and pedigree analyses</td> <td>Identifies local ancestry in windows (source of noise), requires phased data</td> <td>http://lamp.icsi.berkeley.edu/lamp/</td> <td>(Baran et al., 2012)</td> </tr> <tr> <td>PLINK</td> <td>Pedigree, Identity by descent/state</td> <td>Estimating inbreeding and relatedness</td> <td>Allows studying identity by descent and by state. PLINK is a multi-purpose tool, facilitating data analysis within the same software</td> <td>NA</td> <td>http://pngu.mgh.harvard.edu/~purcell/plink/</td> <td>(Purcell et al., 2007)</td> </tr> <tr> <td>VCFTOOLS</td> <td>Pedigree, Identity by descent/state</td> <td>Estimating inbreeding and relatedness</td> <td>Computes unadjusted Ajk and kinship coefficient</td> <td>NA</td> <td>https://vcftools.github.io/man_latest.html</td> <td>(Danecek et al., 2011)</td> </tr> <tr> <td>KING</td> <td>Pedigree, Identity by descent/state</td> <td>Estimating inbreeding and relatedness, multivariate analysis</td> <td>Mendelian error checking, testing family structure, highly accurate kinship coefficient, association analysis, population structure inference</td> <td>Kinship coefficient also computed in VCFTOOLS</td> <td>http://people.virginia.edu/~wc9c/KING/Download.htm</td> <td>(Manichaikul et al., 2010)</td> </tr> <tr> <td>COLONY</td> <td>Pedigrees</td> <td>Pedigree inference from SNPs</td> <td>Robust even with high error rates (e.g. low-depth sequencing). Can handle haplo-diploids systems (e.g. ants). Multi-threaded.</td> <td>Can only simulate genotypes with the Windows version.</td> <td>https://www.zsl.org/science/software/colony</td> <td>(Wang, 2019)</td> </tr> <tr> <td>sequoia</td> <td>Pedigrees</td> <td>Pedigree inference from SNPs</td> <td>Can be applied to large pedigrees (&gt;1000 individuals). Accomodates unknown birth times.</td> <td>Handles hundreds of SNPs. For whole-genome data, preliminary filtering and LD-pruning may be recommended. Efficient with ~100 SNPs.</td> <td>https://cran.r-project.org/web/packages/sequoia/index.html</td> <td>(Huisman, 2017)</td> </tr> <tr> <td>LDHat</td> <td>Recombination</td> <td>Estimating variation in recombination rates along a genome</td> <td>Handles unphased and missing data, underlying model can be used for organisms such as viruses or bacteria</td> <td>Limited to 300 sequences, specific format (not VCF), model for recombination hotspots based on human data</td> <td>http://ldhat.sourceforge.net/</td> <td>(McVean et al., 2002)</td> </tr> <tr> <td>LDHot</td> <td>Recombination</td> <td>Identifying recombination hotspots</td> <td>Specifically designed for detecting recombination hotspots</td> <td>Requires data to be phased, working with LDHat</td> <td>https://github.com/auton1/LDhot</td> <td>(Myers et al., 2005)</td> </tr> <tr> <td>iSMC</td> <td>Recombination</td> <td>Recombination from a single diploid genome</td> <td>No phasing needed. Accepts VCF files as input.</td> <td>Introgression and demographic misspecification may bias results. No detailed tutorial</td> <td>https://github.com/gvbarroso/iSMC</td> <td>(Barroso et al., 2018)</td> </tr> <tr> <td>LDHelmet</td> <td>Recombination</td> <td>Estimating variation in recombination rates along a genome</td> <td>Higher accuracy than LDHat</td> <td>Requires phased data. Does not handle VCF, only fasta and fastq formats. Requires dividing the genome in short segments to be analysed in parallel.</td> <td>https://sourceforge.net/projects/ldhelmet/</td> <td>(Chan et al., 2012)</td> </tr> </tbody> </table> <p>&lt;/div&gt;</p> <div class="caption"> Table 1: Summary of methods dedicated to infer population structure. </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2025 Yann X.C. Bourgeois. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. Photos from <a href="https://unsplash.com" target="_blank" rel="external nofollow noopener">Unsplash</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js?a0db7e5d5c70cc3252b3138b0c91dcaf" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?e0514a05c5c95ac1a93a8dfd5249b92e"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script> <script src="/assets/js/mathjax-setup.js?70d799092f862ad98c7876aa47712e20"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script defer src="/assets/js/progress-bar.js?2f30e0e6801ea8f5036fa66e1ab0a71a" type="text/javascript"></script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>
    addBackToTop();
  </script> <script type="module" src="/assets/js/search/ninja-keys.min.js?a3446f084dcaecc5f75aa1757d087dcf"></script> <ninja-keys hidebreadcrumbs noautoloadmdicons placeholder="Type to start searching"></ninja-keys> <script src="/assets/js/search-setup.js?6c304f7b1992d4b60f7a07956e52f04a"></script> <script src="/assets/js/search-data.js"></script> <script src="/assets/js/shortcut-key.js?6f508d74becd347268a7f822bca7309d"></script> </body> </html>