<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Methods in population genomics | Yann X.C. Bourgeois </title> <meta name="author" content="Yann X.C. Bourgeois"> <meta name="description" content="Ongoing project"> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-table@1.22.4/dist/bootstrap-table.min.css" integrity="sha256-uRX+PiRTR4ysKFRCykT8HLuRCub26LgXJZym3Yeom1c=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" href="/assets/css/scholar-icons.css?62b2ac103a88034e6882a5be5f3e2772"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="/assets/img/palmtreeicon.png?17bc196ea3bf2641d4cdcb8d2cfe16d6"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://yannbourgeois.github.io/projects/4_project/"> <script src="/assets/js/theme.js?9a0c749ec5240d9cda97bc72359a72c0"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>
    initTheme();
  </script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> <span class="font-weight-bold">Yann</span> X.C. Bourgeois </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item active"> <a class="nav-link" href="/projects/">projects <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">repositories </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="nav-item "> <a class="nav-link" href="/teaching/">teaching </a> </li> <li class="nav-item "> <a class="nav-link" href="/people/">people </a> </li> <li class="nav-item dropdown "> <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">submenus </a> <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown"> <a class="dropdown-item " href="/publications/">publications</a> <div class="dropdown-divider"></div> <a class="dropdown-item " href="/projects/">projects</a> <div class="dropdown-divider"></div> <a class="dropdown-item " href="/blog/">blog</a> </div> </li> <li class="nav-item"> <button id="search-toggle" title="Search" onclick="openSearchModal()"> <span class="nav-link">ctrl k <i class="ti ti-search"></i></span> </button> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Methods in population genomics</h1> <p class="post-description">Ongoing project</p> </header> <article> <p>There are many methods in population genomics. To help newcomers in the field, we published a review on the main methods in the field, available <a href="https://doi.org/10.1111/mec.15989" rel="external nofollow noopener" target="_blank">here</a></p> <h2 id="table-1-summary-of-methods-dedicated-to-infer-population-structure">Table 1: Summary of methods dedicated to infer population structure.</h2> <table data-click-to-select="true" data-height="460" data-pagination="true" data-search="true" data-toggle="table" data-url="/assets/json/Table_1.json"> <thead> <tr> <th data-checkbox="true"></th> <th data-field="Software" data-halign="left" data-align="center" data-sortable="true">ID</th> <th data-field="Class of method" data-halign="center" data-align="right" data-sortable="true">Class of method</th> <th data-field="Link" data-halign="right" data-align="left" data-sortable="true">Link</th> </tr> </thead> </table> <h2 id="table-2-summary-of-methods-dedicated-to-detect-selection">Table 2: Summary of methods dedicated to detect selection</h2> <table data-click-to-select="true" data-height="460" data-pagination="true" data-search="true" data-toggle="table" data-url="/assets/json/Table_1.json"> <thead> <tr> <th data-checkbox="true"></th> <th data-field="Software" data-halign="left" data-align="center" data-sortable="true">ID</th> <th data-field="Class of method" data-halign="center" data-align="right" data-sortable="true">Class of method</th> <th data-field="Link" data-halign="right" data-align="left" data-sortable="true">Link</th> </tr> </thead> </table> <h2 id="table-3-summary-of-methods-dedicated-to-reconstruct-demographic-history">Table 3: Summary of methods dedicated to reconstruct demographic history</h2> <table data-click-to-select="true" data-height="460" data-pagination="true" data-search="true" data-toggle="table" data-url="/assets/json/Table_1.json"> <thead> <tr> <th data-checkbox="true"></th> <th data-field="Software" data-halign="left" data-align="center" data-sortable="true">ID</th> <th data-field="Class of method" data-halign="center" data-align="right" data-sortable="true">Class of method</th> <th data-field="Link" data-halign="right" data-align="left" data-sortable="true">Link</th> </tr> </thead> </table> <h2 id="table-4-summary-of-simulators-and-simulation-based-methods-for-selection-and-demographic-inference">Table 4: Summary of simulators and simulation-based methods for selection and demographic inference.</h2> <table data-click-to-select="true" data-height="460" data-pagination="true" data-search="true" data-toggle="table" data-url="/assets/json/Table_1.json"> <thead> <tr> <th data-checkbox="true"></th> <th data-field="Software" data-halign="left" data-align="center" data-sortable="true">ID</th> <th data-field="Class of method" data-halign="center" data-align="right" data-sortable="true">Class of method</th> <th data-field="Link" data-halign="right" data-align="left" data-sortable="true">Link</th> </tr> </thead> </table> <table> <thead> <tr> <th>Software</th> <th>Class of method</th> <th>Purpose</th> <th>Specifics</th> <th>Issues and warnings</th> <th>Link</th> <th>Reference</th> </tr> </thead> <tbody> <tr> <td>POPGenome</td> <td>Differentiation/Diversity/Recombination</td> <td>Computing summary statistics based on AFS and LD along genomes</td> <td>Accepts VCF and GFF/GFT files, efficient and fast. Tests for admixture available (ABBA BABA test). Includes basic coalescence simulations (ms and msms)</td> <td>Mostly limited to summary statistics (but coalescent simulations are possible). No built-in SNP calling module</td> <td>http://catchenlab.life.illinois.edu/stacks/</td> <td>(Pfeifer et al., 2014)</td> </tr> <tr> <td>ANGSD</td> <td>Differentiation/Diversity/Recombination</td> <td>Computing summary statistics based on AFS and LD along genomes</td> <td>Able to process BAM files, built-in procedures for data filtering, admixture analysis. Suited for low-depth data. Includes a suite of methods to estimate relatedness (NGSRelate).</td> <td>Mostly limited to summary statistics. Tutorials not always up-to-date.</td> <td>https://github.com/ANGSD/angsd ; https://github.com/ANGSD/NgsRelate</td> <td>(Korneliussen et al., 2014; Hangh√∏j et al., 2019)</td> </tr> <tr> <td>Arlequin</td> <td>Differentiation/Diversity/Recombination</td> <td>Computing summary statistics based on AFS and LD along genomes</td> <td>Can output AFS for further analysis in fastsimcoal2</td> <td>Slower than PopGenome, requires a specific format and file conversion.</td> <td>http://cmpg.unibe.ch/software/arlequin35/Arl35Downloads.html</td> <td>(Excoffier and Lischer, 2010)</td> </tr> <tr> <td>VCFTOOLS</td> <td>Differentiation/Diversity/Recombination</td> <td>Computing summary statistics based on AFS and LD along genomes</td> <td>Fast. VCFTOOLS can also be used for SNP filtering</td> <td>Less summary statistics than POPGenome</td> <td>https://vcftools.github.io/man_latest.html</td> <td>(Danecek et al., 2011)</td> </tr> <tr> <td>ATLAS</td> <td>Differentiation/Diversity/Recombination</td> <td>Low depth sequencing/ancient samples analsis</td> <td>Particularly suited for analyzing ancient samples. Includes sets of tools to call variants, estimate post-mortem damage, inbreeding, genetic diversity. Produces the input file for PSMC (demography from a single diploid genome)</td> <td>Better used in combination with GATK pipelines. Still in development</td> <td>https://bitbucket.org/wegmannlab/atlas/wiki/Home</td> <td>(Link et al., 2017)</td> </tr> <tr> <td>POPTREE2</td> <td>Genetic differentiation</td> <td>Visualizing a matrix of pairwise differentiation statistics as a tree</td> <td>Can be used for pooled datasets, several statistics can be used</td> <td>Differentiation measures alone do not necessarily retrieve the actual history of populations</td> <td>http://www.med.kagawa-u.ac.jp/~genomelb/takezaki/poptree2/index.html</td> <td>(Takezaki et al., 2010)</td> </tr> <tr> <td>EEMS</td> <td>Landscape genomics</td> <td>Estimating barriers to gene flow in a spatial context</td> <td>Estimates pairwise relatedness between all samples, and compares it to isolation-by-distance expectations to identify barriers to gene flow and corridors of higher connectivity. Can handle both haploid and diploid data</td> <td>Requires to convert VCF file into PLINK binary format. Estimates effective migration rates (does not disantangle migration rates and effective population sizes). Setting parameters for the MCMC chain requires some trial-and-error</td> <td>https://github.com/dipetkov/eems</td> <td>(Petkova et al., 2015)</td> </tr> <tr> <td>MAPS</td> <td>Landscape genomics</td> <td>Estimating barriers to gene flow in a spatio-temporal context</td> <td>Expands on EEMS, but takes into account the phase to reconstruct past changes in connectivity. Can disentantle migration rates and effective population sizes (unlike EEMS)</td> <td>Relies on identity-by-descent tracks, requiring phasing (for example using BEAGLE). A pipeline to obtain IBD tracks is available, with a few details here: https://github.com/halasadi/ibd_data_pipeline/issues/1</td> <td>https://github.com/halasadi/MAPS</td> <td>(Al-Asadi et al., 2019)</td> </tr> <tr> <td>TESS3R</td> <td>Landscape genomics</td> <td>Grouping individuals in clusters maximizing HW equilibrium and LD between loci</td> <td>Incorporates geographic information of samples. Can run genome scans of selection based on contrasting ancestral and modern allele frequencies.</td> <td>Importing data requires using conversion tools found in the LEA suite</td> <td>https://bcm-uga.github.io/TESS3_encho_sen/</td> <td>(Caye et al., 2016)</td> </tr> <tr> <td>SNPRelate</td> <td>Multivariate analysis</td> <td>Summarizing variance across loci and visualizing inter-individual genetic distance</td> <td>Fast. Can use VCF files as an input</td> <td>Requires careful interpretation (Jombard et al. 2009)</td> <td>https://bioconductor.org/packages/release/bioc/html/SNPRelate.html</td> <td>(Zheng et al., 2012)</td> </tr> <tr> <td>Eigenstrat/smartpca</td> <td>Multivariate analysis</td> <td>Summarizing variance across loci and visualizing inter-individual genetic distance</td> <td>Fast. Can use VCF files as an input</td> <td>Requires careful interpretation (Jombard et al. 2009)</td> <td>https://github.com/DReichLab/EIG/tree/master/EIGENSTRAT</td> <td>(Price et al., 2006)</td> </tr> <tr> <td>DAPC (adegenet)</td> <td>Multivariate analysis/Clustering</td> <td>Maximizes divergence between groups identified by PCA</td> <td>Fast. Less sensitive to HWE assumptions. Claims to be more efficient than Structure</td> <td>Requires careful interpretation (Jombard et al. 2009)</td> <td>http://adegenet.r-forge.r-project.org/</td> <td>(Jombart et al., 2010)</td> </tr> <tr> <td>sPCA (adegenet)</td> <td>Multivariate analysis/Clustering</td> <td>Spatially explicit model to assess population structure</td> <td>Spatially explicit and able to detect cryptic structure. Fast.</td> <td>Does not take into account HW equilibrium or LD</td> <td>http://adegenet.r-forge.r-project.org/</td> <td>(Jombart et al., 2008)</td> </tr> <tr> <td>LAMP</td> <td>Pedigree, Identity by descent/state</td> <td>Chromosome painting, relatedness</td> <td>LAMP also allows for association and pedigree analyses</td> <td>Identifies local ancestry in windows (source of noise), requires phased data</td> <td>http://lamp.icsi.berkeley.edu/lamp/</td> <td>(Baran et al., 2012)</td> </tr> <tr> <td>PLINK</td> <td>Pedigree, Identity by descent/state</td> <td>Estimating inbreeding and relatedness</td> <td>Allows studying identity by descent and by state. PLINK is a multi-purpose tool, facilitating data analysis within the same software</td> <td>NA</td> <td>http://pngu.mgh.harvard.edu/~purcell/plink/</td> <td>(Purcell et al., 2007)</td> </tr> <tr> <td>VCFTOOLS</td> <td>Pedigree, Identity by descent/state</td> <td>Estimating inbreeding and relatedness</td> <td>Computes unadjusted Ajk and kinship coefficient</td> <td>NA</td> <td>https://vcftools.github.io/man_latest.html</td> <td>(Danecek et al., 2011)</td> </tr> <tr> <td>KING</td> <td>Pedigree, Identity by descent/state</td> <td>Estimating inbreeding and relatedness, multivariate analysis</td> <td>Mendelian error checking, testing family structure, highly accurate kinship coefficient, association analysis, population structure inference</td> <td>Kinship coefficient also computed in VCFTOOLS</td> <td>http://people.virginia.edu/~wc9c/KING/Download.htm</td> <td>(Manichaikul et al., 2010)</td> </tr> <tr> <td>COLONY</td> <td>Pedigrees</td> <td>Pedigree inference from SNPs</td> <td>Robust even with high error rates (e.g. low-depth sequencing). Can handle haplo-diploids systems (e.g. ants). Multi-threaded.</td> <td>Can only simulate genotypes with the Windows version.</td> <td>https://www.zsl.org/science/software/colony</td> <td>(Wang, 2019)</td> </tr> <tr> <td>sequoia</td> <td>Pedigrees</td> <td>Pedigree inference from SNPs</td> <td>Can be applied to large pedigrees (&gt;1000 individuals). Accomodates unknown birth times.</td> <td>Handles hundreds of SNPs. For whole-genome data, preliminary filtering and LD-pruning may be recommended. Efficient with ~100 SNPs.</td> <td>https://cran.r-project.org/web/packages/sequoia/index.html</td> <td>(Huisman, 2017)</td> </tr> <tr> <td>LDHat</td> <td>Recombination</td> <td>Estimating variation in recombination rates along a genome</td> <td>Handles unphased and missing data, underlying model can be used for organisms such as viruses or bacteria</td> <td>Limited to 300 sequences, specific format (not VCF), model for recombination hotspots based on human data</td> <td>http://ldhat.sourceforge.net/</td> <td>(McVean et al., 2002)</td> </tr> <tr> <td>LDHot</td> <td>Recombination</td> <td>Identifying recombination hotspots</td> <td>Specifically designed for detecting recombination hotspots</td> <td>Requires data to be phased, working with LDHat</td> <td>https://github.com/auton1/LDhot</td> <td>(Myers et al., 2005)</td> </tr> <tr> <td>iSMC</td> <td>Recombination</td> <td>Recombination from a single diploid genome</td> <td>No phasing needed. Accepts VCF files as input.</td> <td>Introgression and demographic misspecification may bias results. No detailed tutorial</td> <td>https://github.com/gvbarroso/iSMC</td> <td>(Barroso et al., 2018)</td> </tr> <tr> <td>LDHelmet</td> <td>Recombination</td> <td>Estimating variation in recombination rates along a genome</td> <td>Higher accuracy than LDHat</td> <td>Requires phased data. Does not handle VCF, only fasta and fastq formats. Requires dividing the genome in short segments to be analysed in parallel.</td> <td>https://sourceforge.net/projects/ldhelmet/</td> <td>(Chan et al., 2012)</td> </tr> <tr> <td>pyrho</td> <td>Recombination</td> <td>Estimating variation in recombination rates along a genome</td> <td>Fast. Takes both phased and unphased data as input. Can take into account the demographic history (to be inferred with, e.g. MSMC2)</td> <td>Outputs a recombination rate (r), but internally calculates rho (4xNexr) and rescales by the provided Ne. Uncertainties in Ne propagate on r.</td> <td>https://github.com/popgenmethods/pyrho</td> <td>(Spence and Song, 2019)</td> </tr> </tbody> </table> <div class="caption"> Table 1: Summary of methods dedicated to infer population structure. </div> <table> <thead> <tr> <th>Software</th> <th>Class of method</th> <th>Purpose</th> <th>Specifics</th> <th>Issues and warnings</th> <th>Link</th> <th>Reference</th> </tr> </thead> <tbody> <tr> <td>ARGWeaver/ARGweaver-D</td> <td>Ancestral Recombination Graphs/coalescence</td> <td>Retracing the whole process of recombination and coalescence along a genome</td> <td>Provides quantitative estimates for TMRCA and topologies at each locus. ARGWeaver-D can estimate introgression. Estimates effective population size. Provides tools to extract summary statistics for the topologies retrieved. Does not require phasing (but slower).</td> <td>High computing cost. Slower on unphased or low depth data. ARGWeaver-D is not part of the Anaconda (Python) distribution (http://compgen.cshl.edu/ARGweaver/doc/argweaver-d-manual.html)</td> <td>Can be installed via conda: conda install -c genomedk argweaver and https://github.com/mjhubisz/argweaver and http://compgen.cshl.edu/ARGweaver/doc/argweaver-d-manual.html</td> <td>(Rasmussen et al., 2014; Hubisz et al., 2020)</td> </tr> <tr> <td>GAPIT3</td> <td>Association</td> <td>Detecting association with environmental/phenotypical features</td> <td>Includes most methods for GWAS studies, including procedures for fast computation, mixed linear models, efficient mixed model association, bayesian methods such as BLINK, diagnostics such as QQ plots and genotype filtering.</td> <td>May be slow for very large datasets</td> <td>https://github.com/jiabowang/GAPIT3</td> <td>(Wang and Zhang, 2020)</td> </tr> <tr> <td>GEMMA</td> <td>Association</td> <td>Detecting association with environmental/phenotypical features</td> <td>Computationnally efficient for large scale datasets</td> <td>Imports data from PLINK format</td> <td>http://www.xzlab.org/software.html</td> <td>(Zhou and Stephens, 2012)</td> </tr> <tr> <td>GENABEL</td> <td>Association</td> <td>Detecting association with environmental/phenotypic features</td> <td>Modularity, facilitates correction for population structure/relatedness.</td> <td>Imports data from PLINK format. No longer supported!</td> <td>http://www.genabel.org/</td> <td>(Aulchenko et al., 2007)</td> </tr> <tr> <td>PLINK</td> <td>Association</td> <td>Detecting association with environmental/phenotypical features</td> <td>Handles a variety of tests for population structure and relatedness</td> <td>Population structure/kinship need to be assessed prior association analysis</td> <td>http://pngu.mgh.harvard.edu/~purcell/plink/</td> <td>Purcell et al., 2007)</td> </tr> <tr> <td>Trinculo</td> <td>Association</td> <td>Detecting association with environmental/phenotypical features</td> <td>Specifically designed to handle categorical variables with more than 2 categories. Performs multinomial logistic regression and provides frequentist and bayesian frameworks.</td> <td>Requires lapack library in Unix. Allows fine-mapping by testing for corrrelations between adjacent markers.</td> <td>https://sourceforge.net/projects/trinculo/</td> <td>(Jostins and McVean, 2016)</td> </tr> <tr> <td>SAMBADA</td> <td>Association/Environmental association</td> <td>Detecting association with environmental/phenotypical features</td> <td>Designed to be fast, underlying models have been kept simple. Allows conversion from PLINK format. Takes into account spatial autocorrelation of individual genotypes. Allows correction for population structure</td> <td>Does not work with pooled data. Possibly high levels of false positives. Relatedness between samples should be assessed independently. Should be used in combination with LFMM or BayPass.</td> <td>http://lasig.epfl.ch/sambada</td> <td>(Stucki et al., 2016)</td> </tr> <tr> <td>Relate</td> <td>Coalescence with recombination</td> <td>Reconstruct genome-wide genealogies for hundreds of samples</td> <td>Provides quantitative estimates for TMRCA and topologies at each locus. Infers past demography (similar to PSMC methods). Infers changes in mutation rates. Performs scans for positive selection over discrete time periods.</td> <td>Requires an outgroup to polarize alleles as ancestral/derived. Requires a recombination map. Does not reconstruct ARG sensus stricto, and does not estimate uncertainty of the local genealogies</td> <td>https://myersgroup.github.io/relate/index.html</td> <td>(Speidel et al., 2019)</td> </tr> <tr> <td>diCal-IBD</td> <td>Coalescent with recombination/IBD</td> <td>Predicting IBD tracts from demographic models</td> <td>High IBD sharing suggests recent positive selection.</td> <td>Uses diCal output to obtain expectations based on demographic scenarios</td> <td>https://sourceforge.net/projects/dical-ibd/</td> <td>(Tataru et al., 2014)</td> </tr> <tr> <td>VolcanoFinder</td> <td>Composite likelihood test</td> <td>Adaptive introgression</td> <td>Detects a specific signature of increase then drop in diversity near a selected locus brought in a population through introgression</td> <td>Private input format. Computationnally intensive, needs to be run in parallel.</td> <td>http://degiorgiogroup.fau.edu/vf.html</td> <td>(Setter et al., 2020)</td> </tr> <tr> <td>SCCT</td> <td>Conditional coalescent tree</td> <td>Detecting positive selection</td> <td>Designed for detecting recent positive selection. Clains to be more precise at identifying selected sites</td> <td>The ancestral state of alleles must be obtained through an outgroup</td> <td>https://github.com/wavefancy/scct</td> <td>(Wang et al., 2014)</td> </tr> <tr> <td>LFMM2</td> <td>Environmental association</td> <td>Detecting adaptation to environmental features</td> <td>Corrects for population structure (and other non-measured systematic biases) using latent factors, faster than BAYENV for large datasets</td> <td>Only performs association with environment. Implemented in the LEA package. Requires imputation (no missing data).</td> <td>http://membres-timc.imag.fr/Olivier.Francois/LEA/tutorial.htm</td> <td>(Caye et al. 2019; Frichot et al., 2013)</td> </tr> <tr> <td>CLUES</td> <td>Genealogies at selected loci</td> <td>Estimate the time at which a beneficial allele rises in frequency</td> <td>Previous version used ARGWeaver output, current version uses Relate. Provides scripts to plot the trajectory of selected alleles.</td> <td>Assumes a panmictic population, neglects the effects of selection at linked sites.</td> <td>https://github.com/35ajstern/clues</td> <td>(Stern et al., 2019)</td> </tr> <tr> <td>PALM</td> <td>Genealogies at selected loci</td> <td>Estimate the strength and timing of selection on polygenic traits</td> <td>Uses genealogies estimated from Relate and results from GWAS to estimate timing and strength of selection for polygenic traits. Should be robust to pleiotropy and residual structure in GWAS</td> <td>May overestimate selection for older events. Only tested in humans.</td> <td>https://github.com/35ajstern/palm</td> <td>(Stern et al., 2020)</td> </tr> <tr> <td>startmrca</td> <td>Genealogies at selected loci</td> <td>Estimate the time at which a beneficial allele rises in frequency</td> <td>Compares genealogies between carriers and non-carriers of an advantageous mutation, assuming a star- genealogy at selected loci. Can handle VCF files</td> <td>Requires a reference panel of noncarrier haplotypes. Sensitive to loca diversity before the sweep, and to migration events during a sweep. More indicated for recent sweeps.</td> <td>https://github.com/jhavsmith/startmrca</td> <td>Smith, Coop, Stephens, &amp; Novembre, 2018)</td> </tr> <tr> <td>Ancestry_HMM-S¬†</td> <td>Identity-by-state tracts</td> <td>Adaptive introgression</td> <td>Estimates the selective coefficient of the introgressed loci through a hidden-Markov chain approach.</td> <td>Requires the time and extent of introgression to be defined by the user</td> <td>https://github.com/jesvedberg/Ancestry_HMM-S/</td> <td>(Svedberg‚Äã et al., 2020)</td> </tr> <tr> <td>H12 test</td> <td>LD</td> <td>Detecting selection using signatures of high LD</td> <td>Does not require phased data. Designed for detecting soft sweeps</td> <td>Coalescent simulations are recommended to evaluate the likelihood of selection</td> <td>https://github.com/ngarud/SelectionHapStats/</td> <td>(Garud et al., 2015)</td> </tr> <tr> <td>LDna</td> <td>LD</td> <td>Detecting selection using signatures of high LD</td> <td>Can be used to address population structure or detect large inversions or indel polymorphism through LD</td> <td>The user needs to play with parameters to ensure robustness of SNPs significantly linked</td> <td>https://github.com/petrikemppainen/LDna</td> <td>(Kemppainen et al., 2015)</td> </tr> <tr> <td>rehh</td> <td>LD</td> <td>Detecting selection using signatures of high LD</td> <td>Can compute both XP-EHH and Rsb. Handles several input formats</td> <td>Requires phased data and high density of markers</td> <td>https://cran.r-project.org/web/packages/rehh/index.html</td> <td>(Gautier and Vitalis, 2012)</td> </tr> <tr> <td>Scan for epistatic interaction (based on LD)</td> <td>LD</td> <td>Polygenic selection/Epistatic interactions</td> <td>Uses genome-wide LD between a candidate locus and the rest of the genomes to identify epistatic interactions. Can test SNP-SNP interaction, or between genomic windows (summarizes genotypes through PCA)</td> <td>Lack of a detailed tutorial</td> <td>https://github.com/leaboyrie/LD_corpc1</td> <td>(Boyrie et al., 2020)</td> </tr> <tr> <td>Selscan</td> <td>LD</td> <td>Detecting selection using signatures of high LD</td> <td>Includes the nSL statistics dedicated to soft sweep detection</td> <td>Does not include utilities to specify the ancestral state of alleles. Requires phased data and high density of markers</td> <td>https://github.com/szpiech/selscan</td> <td>(Szpiech and Hernandez, 2014)</td> </tr> <tr> <td>BALLET</td> <td>Likelihood test for balancing selection</td> <td>Detecting balancing selection</td> <td>Designed for detecting ancient balancing selection. Does not require phasing</td> <td>Requires whole-genome data and recombination map. The ancestral state of alleles must be obtained through an outgroup</td> <td>http://www.personal.psu.edu/mxd60/ballet.html</td> <td>(DeGiorgio et al., 2014)</td> </tr> <tr> <td>Betascan2</td> <td>Local associations of allele frequencies</td> <td>Detecting balancing selection</td> <td>Uses correlations in frequencies between genomically proximate SNPs to compute a score. Can incorporate information about ancestral/derived alleles, fixed derived variants and normalizes the statistics depending on the amount of sites in a given genomic window. Very detailed tutorial and utilities.</td> <td>Requires estimating the length distribution of ancestral fragments on each side of the selected site. The 95% percentile can be estimated with the formula L=-log(0.05)/(T x rho), with T the time since selection in generations and rho the effective recombination rate/ generation.</td> <td>https://github.com/ksiewert/BetaScan</td> <td>(Siewert and Voight, 2017, 2020)</td> </tr> <tr> <td>NCD statistics</td> <td>Local associations of allele frequencies</td> <td>Detecting balancing selection</td> <td>Examines the observed and expected frequency spectra of polymorphisms in genomic windows to test for selection. Can incorporate fixed differences with an outgroup (NCD2), but not mandatory (NCD1)</td> <td>Private input format, requires simulations to calibrate the statistics. Requires to define the expected equilibrium frequency of alleles (usually between 0.3 and 0.5). Low sensitivity below these frequencies.</td> <td>https://github.com/bbitarello/NCD-Statistics</td> <td>(Bitarello et al., 2018)</td> </tr> <tr> <td>Bayescan</td> <td>Population differentiation</td> <td>Detecting positive selection and local adaptation</td> <td>Incorporates uncertainty on allele frequencies due to low sample sizes</td> <td>Sensitive to priors on the ratio of selected/neutral sites. False positive rates can be high under scenarios of demographic expansion, admixture and isolation by distance</td> <td>http://cmpg.unibe.ch/software/BayeScan/</td> <td>(Foll and Gaggiotti, 2008)</td> </tr> <tr> <td>FDIST2</td> <td>Population differentiation</td> <td>Detecting positive selection and local adaptation</td> <td>Allows to control for hierarchical population structure</td> <td>False positive rate is high when an island model cannot be assumed</td> <td>http://datadryad.org/resource/doi:10.5061/dryad.v8d05</td> <td>(Beaumont and Balding, 2004)</td> </tr> <tr> <td>PCAdapt</td> <td>Population differentiation</td> <td>Detecting positive selection and local adaptation</td> <td>Does not require to define populations. Handles admixed populations and pooled datasets</td> <td>False positive rate can be high</td> <td>http://membres-timc.imag.fr/Michael.Blum/PCAdapt.html</td> <td>(Duforet-Frebourg et al., 2016)</td> </tr> <tr> <td>SelEstim</td> <td>Population differentiation</td> <td>Detecting positive selection and local adaptation</td> <td>Can estimate the coefficients of selection. Calibration using a pseudo-observed dataset (can be used in combination with the R function simulate.baypass() in BayPass).</td> <td>Assumes a Wrigth-Fisher island model.</td> <td>http://www1.montpellier.inra.fr/CBGP/software/selestim/</td> <td>(Vitalis et al., 2014)</td> </tr> <tr> <td>Bayenv, BayPass</td> <td>Population differentiation/Association</td> <td>Detecting positive selection and adaptation to environmental features</td> <td>Less sensitive to population demographic history than previous methods. Handle pooled datasets</td> <td>Significance thresholds need to be determined from pseudo-observed datasets. Calibration with neutral SNPs is recommended. BayPass better estimates the kinship matrix</td> <td>http://www1.montpellier.inra.fr/CBGP/software/baypass/ ; https://bitbucket.org/tguenther/bayenv2_public/src</td> <td>(G√ºnther and Coop, 2013; Gautier, 2015)</td> </tr> <tr> <td>FLK</td> <td>Population differentiation/Association</td> <td>Detecting positive selection and local adaptation</td> <td>Less sensitive to population demographic history than previous methods</td> <td>Requires an outgroup population</td> <td>https://qgsp.jouy.inra.fr/index.php?option=com_content&amp;view=article&amp;id=50&amp;Itemid=55</td> <td>(Bonhomme et al., 2010)</td> </tr> <tr> <td>LSD</td> <td>Population differentiation/Population-branch test</td> <td>Detecting positive selection and local adaptation</td> <td>Compares the level of exclusively shared differences between internal and external branches of a population tree. Allows testing selection occurring on the ancestral branch leading to two populations.</td> <td>Requires several populations to perform the test. May be less sensitive to selection on standing variation.</td> <td>https://bitbucket.org/plibrado/LSD</td> <td>(Librado and Orlando, 2018)</td> </tr> <tr> <td>POPBAM</td> <td>Summary statistics</td> <td>Detecting selection using AFS, differentiation</td> <td>Extracts summary statistics directly from BAM files</td> <td>Does not allow for sophisticated filtering and SNP calling</td> <td>http://popbam.sourceforge.net/</td> <td>(Garrigan, 2013)</td> </tr> <tr> <td>VCFTOOLS</td> <td>Summary statistics</td> <td>Detecting selection using AFS, differentiation</td> <td>Extracts summary statistics from VCF files. Also allows VCF filtering and conversion</td> <td>Set of summary statistics not as extensive as PopGenome</td> <td>http://vcftools.sourceforge.net/</td> <td>(Danecek et al., 2011)</td> </tr> <tr> <td>RAiSD</td> <td>Summary statistics/Allele frequency spectrum + LD</td> <td>Detecting positive selection and local adaptation</td> <td>Scans the genome for composite signals of selective sweeps summarized by the Œº statistics. Corrects for the effects of background selection by estimating a threshold value for the statistics based on simulations with background selection</td> <td>Uses a single population of interest.</td> <td>https://github.com/alachins/raisd</td> <td>(Alachiotis and Pavlidis, 2018)</td> </tr> <tr> <td>TASSEL</td> <td>Summary statistics/Association</td> <td>Detecting association with phenotype</td> <td>User friendly (Java interface), corrects for relatedness, allows computing summary statistics (LD, diversity)</td> <td>Requires relatedness to be assessed externally (with e.g. STRUCTURE)</td> <td>http://www.maizegenetics.net/tassel</td> <td>(Bradbury et al., 2007)</td> </tr> <tr> <td>ANGSD</td> <td>Summary statistics/Association/Population Branch test</td> <td>Detecting selection using AFS, differentiation, association with functional traits</td> <td>Allows for association using generalized linear models</td> <td>Descriptive statistics. P-values need to be evaluated through coalescent simulations.</td> <td>http://www.popgen.dk/angsd/index.php/ANGSD</td> <td>(Korneliussen et al., 2014)</td> </tr> <tr> <td>SweeD</td> <td>Summary statistics/Composite Likelihood test</td> <td>Designed for whole genome data (or large continuous regions)</td> <td>Supports Fasta and VCF formats. Estimates selection coefficients.</td> <td>NA</td> <td>http://pop-gen.eu/wordpress/software/sweed</td> <td>(Degiorgio et al., 2016)</td> </tr> <tr> <td>selectionTools</td> <td>Summary statistics/LD</td> <td>Detecting selection using AFS, differentiation and LD statistics</td> <td>Allows combining several tools in a single pipeline. Includes phasing tools.</td> <td>Set of available summary statistics remains limited (same as VCFtools + Fay and Wu‚Äôs H)</td> <td>https://github.com/MerrimanLab/selectionTools</td> <td>(Cadzow et al., 2014)</td> </tr> <tr> <td>PAML/CODEML</td> <td>Summary statistics/phylogeny</td> <td>Distribution of fitness effects/selection on coding variation</td> <td>Estimates selection along branches in a phylogeny for genes of interest, contrasting patterns of synonymous and non-synonymous substitutions. A detailed tutorial is available here: https://link.springer.com/protocol/10.1007%2F978-1-4939-1438-8_4#Sec29</td> <td>Slow for large datasets. Needs to be parallelised.</td> <td>http://abacus.gene.ucl.ac.uk/software/paml.html</td> <td>(Yang, 2007)</td> </tr> <tr> <td>polyDFE2.0</td> <td>Summary statistics/phylogeny</td> <td>Distribution of fitness effects/selection on coding variation</td> <td>Can test for invariance of DFEs across datasets (genomic regions within species, or different species). No need for divergence estimates (does not assume that the same DFE is shared between species and outgroup). Very detailed tutorial available here: https://link.springer.com/protocol/10.1007/978-1-0716-0199-0_6</td> <td>Comparisons require a large number of SNPs for each dataset for comparisons to be meaningful</td> <td>https://github.com/paula-tataru/polyDFE</td> <td>(Tataru and Bataillon, 2019)</td> </tr> <tr> <td>POPGenome</td> <td>Summary statistics/Population Branch test</td> <td>Detecting selection using AFS, differentiation</td> <td>Fast, embedded in R, allows using annotation files (GFF/GTF format).</td> <td>Does not perform association, but can be used in combination with GENABEL within R</td> <td>https://cran.r-project.org/web/packages/PopGenome/index.html</td> <td>(Pfeifer et al., 2014)</td> </tr> </tbody> </table> <div class="caption"> Table 2: Summary of methods dedicated to detect selection </div> <table> <thead> <tr> <th>Software</th> <th>Class of method</th> <th>Purpose</th> <th>Specifics</th> <th>Issues and warnings</th> <th>Link</th> <th>Reference</th> </tr> </thead> <tbody> <tr> <td>Dsuite</td> <td>ABBA-BABA</td> <td>Identifying past events of admixture between populations</td> <td>Fast, handles VCF format. Suited for low-depth sequencing (handles uncertainties on genotypes). Provides a set of summary statistics that are useful to investigate complex admixture events</td> <td>Requires an outgroup sequence. The methods cannot estimate the direction of gene flow.</td> <td>https://github.com/millanek/Dsuite</td> <td>(Malinsky et al., 2020)</td> </tr> <tr> <td>RENT+</td> <td>Ancestral Recombination Graphs/coalescence</td> <td>Retracing the whole process of recombination and coalescence along a genome</td> <td>Faster than first version of ARGWeaver.</td> <td>Requires phased haplotypes. Specific input format. No built-in functions to extract information from genealogies.</td> <td>https://github.com/SajadMirzaei/RentPlus</td> <td>(Mirzaei and Wu, 2017)</td> </tr> <tr> <td>TREEMIX</td> <td>Clustering and characterizing admixture</td> <td>Admixture graph, infers most likely admixture events in a tree</td> <td>Based on allele frequencies and can be used for pooled data.</td> <td>Requires multiple runs to properly assess the likelihood of each model</td> <td>https://bitbucket.org/nygcresearch/treemix/src</td> <td>(Pickrell and Pritchard, 2012)</td> </tr> <tr> <td>G-PhoCS</td> <td>Coalescence/Bayesian</td> <td>Estimating population divergence and migration parameters using a coalescent framework</td> <td>Bayesian + MCMC, handles ancient samples</td> <td>Parameters scaled by mutation rate, no admixture</td> <td>http://compgen.cshl.edu/GPhoCS/</td> <td>(Gronau et al., 2011)</td> </tr> <tr> <td>IMa3</td> <td>Coalescence/Bayesian</td> <td>Inferring parameters from an isolation with migration (IM) model</td> <td>Fully bayesian approach, can perform joint estimates of parameters in L-mode and test for nested models. Can estimate phylogenetic relationships and migration rates</td> <td>IM model is the only one available. Discrete admixture cannot be tested. Can only use subsets of whole-genome resequencing data. Recent splits lead to overestimate migration rates</td> <td>https://github.com/jodyhey/IMa3</td> <td>(Hey and Nielsen, 2007)</td> </tr> <tr> <td>ABLE</td> <td>Coalescence/Composite Likelihood</td> <td>Model comparison and parameters estimation</td> <td>Uses both allele frequency spectrum and linkage disequilibrium within blocks of a pre-specified size.</td> <td>Relies on ms syntax. Determining the most informative size for blocks requires performing pilot runs.</td> <td>https://github.com/champost/ABLE</td> <td>(Beeravolu et al., 2016)</td> </tr> <tr> <td>Stairway2</td> <td>Coalescence/Composite Likelihood</td> <td>Inferring change in Ne with time</td> <td>User-friendly. Fast. Suitable for pools or low-depth sequencing.</td> <td>Cannot handle migration or population splits.</td> <td>https://github.com/xiaoming-liu/stairway-plot-v2</td> <td>(Liu and Fu, 2020)</td> </tr> <tr> <td>fastsimcoal2</td> <td>Coalescence/Likelihood</td> <td>Model comparison and parameters estimation</td> <td>Performs coalescent simulations, parameter estimation and model testing using a fast likelihood method. Can handle arbitrarily complex scenarios for any type of marker</td> <td>The maximum-likelihood method only uses the allele frequency spectrum. Several runs (20-100) are needed to explore the likelihood space.</td> <td>http://cmpg.unibe.ch/software/fastsimcoal2/</td> <td>(Excoffier et al., 2013)</td> </tr> <tr> <td>‚àÇa‚àÇi</td> <td>Diffusion approximation of the AFS</td> <td>Model comparison and parameters estimation</td> <td>Run time does not depend on the number of SNPs included, does not require coalescent simulations, handles arbitrarily complex scenarios. Fast estimation of confidence intervals around parameters estimates (Godambe method). Suitable for pools/low-depth sequencing</td> <td>requires some knowledge of Python. Limited to 3 populations. Several runs (20-100) are needed to explore the likelihood space.</td> <td>https://bitbucket.org/gutenkunstlab/dadi</td> <td>(Gutenkunst et al., 2009)</td> </tr> <tr> <td>moments</td> <td>Diffusion approximation of the AFS</td> <td>Model comparison and parameters estimation</td> <td>Based on Python, syntax similar to ‚àÇa‚àÇi. Can handle selection. Can use VCF files as input.</td> <td>Requires some knowledge of Python. Limited to 5 populations. Several runs (20-100) are needed to explore the likelihood space.</td> <td>https://bitbucket.org/simongravel/moments/src/master/</td> <td>(Jouganous et al., 2017)</td> </tr> <tr> <td>momi2</td> <td>Diffusion approximation of the AFS</td> <td>Model comparison and parameters estimation</td> <td>Can scale to ten populations. Can simulate and read data in the VCF format. Detailed tutorials available</td> <td>Does not handle continuous gene flow</td> <td>https://github.com/popgenmethods/momi2</td> <td>(Kamm et al., 2020)</td> </tr> <tr> <td>KIMTree</td> <td>Diffusion approximation/Bayesian</td> <td>Estimating divergence time between populations and testing for topologies. Estimate divergence times and past effective sex-ratio along branches of a populations tree.</td> <td>Fast and user-friendly. R scripts to obtain plots are available. Suitable for pools/low-depth sequencing. The method is conditional on a prior topology provided by the user. It computes DIC for a given topology, allowing to test for the best one.</td> <td>Strong selection on the sex chromosome can produce male-biased sex-ratios. Times are given in diffusion time scale, and can be converted in demographic times using independent estimates of Ne.</td> <td>http://www1.montpellier.inra.fr/CBGP/software/kimtree/download.html</td> <td>(Clemente et al., 2018)</td> </tr> <tr> <td>GADMA</td> <td>Genetic algorithm</td> <td>Model comparison and parameters estimation</td> <td>Based on moments and ‚àÇa‚àÇi. Automates the search for the best set of models explaining a given frequency spectrum.</td> <td>Limited to three populations at the moment.</td> <td>https://github.com/ctlab/GADMA</td> <td>(Noskova et al., 2020)</td> </tr> <tr> <td>DoRIS</td> <td>Identity by Descent (IBD) tract</td> <td>Testing various demographic scenario</td> <td>Uses variation in IBD tracts length to test for various demographic models.</td> <td>IBD must be inferred first with, e.g., BEAGLE. Handles a limited set of demographic scenarios. Modification in the code is required for more complex scenarios</td> <td>https://github.com/pierpal/DoRIS</td> <td>(Palamara and Pe‚Äôer, 2013)</td> </tr> <tr> <td>Unnamed.</td> <td>Identity by state (IBS) tract</td> <td>Predict observed patterns of Identity by state along a genome by fittingan appropriate, arbitrary complex demographic model</td> <td>Allows bootstrapping and estimating confidence over parameter estimates with ms</td> <td>Specific input format (similar to MSMC or ARGWeaver)</td> <td>https://github.com/kelleyharris/Inferring-demography-from-IBS</td> <td>(Harris and Nielsen, 2013)</td> </tr> <tr> <td>ASTRAL-2</td> <td>Phylogeny</td> <td>Builds species trees using short non-recombining sequences</td> <td>Coalescence-based. Suitable for short loci (e.g. RAD-seq and GBS)</td> <td>More reliable under high incomplete lineage sorting that SVDQuartets and NJst (Chou et al. 2015)</td> <td>https://github.com/smirarab/ASTRAL</td> <td>(Mirarab and Warnow, 2015)</td> </tr> <tr> <td>BEAST2</td> <td>Phylogeny</td> <td>Network reconstruction and phylogenetic relationships</td> <td>User friendly. Can be used to track changes in effective population sizes (Bayesian Skyline Plots). Possible to estimate divergence times</td> <td>Slow for large datasets. Requires sequence data that can be produced by , e.g., Stacks for RAD-seq data</td> <td>http://beast2.org/</td> <td>(Drummond and Rambaut, 2007; Bouckaert et al., 2014)</td> </tr> <tr> <td>IQ-Tree 2</td> <td>Phylogeny</td> <td>Divergence time estimation and phylogenetic relationships</td> <td>User-friendly, can be run locally or on a webserver, very detailed tutorials. Fast and accurate.</td> <td>Still no tutorial for analyzing big data (last checked December 2020).</td> <td>http://www.iqtree.org/</td> <td>(Minh et al., 2020)</td> </tr> <tr> <td>MCMCTree and MCMCTreeR</td> <td>Phylogeny</td> <td>Divergence time estimation and phylogenetic relationships</td> <td>Included in PAML. A R program is designed to help choosing relevant priors and interpreting results https://github.com/PuttickMacroevolution/MCMCtreeR</td> <td>Bayesian, sensitive to priors. Requires a resolved phylogeny and an alignment. Slow for large datasets. Not suited for recent divergence and high gene flow.</td> <td>http://abacus.gene.ucl.ac.uk/software/paml.html</td> <td>(Yang, 2007; Puttick, 2019)</td> </tr> <tr> <td>NJst</td> <td>Phylogeny</td> <td>Builds species trees using short non-recombining sequences</td> <td>Available in the R package phybase. Estimates populations/species tree from gene trees</td> <td>Requires splitting part of the genome into non-recombining ‚Äúloci‚Äù.</td> <td>https://github.com/bomeara/phybase/</td> <td>(Liu and Yu, 2010, 2011)</td> </tr> <tr> <td>PHRAPL</td> <td>Phylogeny</td> <td>Admixture graph, reticulated evolution</td> <td>Uses trees in the NEWICK format as an input to infer topology, migration rates, divergence times. Similar to ABC in spirit, using tree topology as a summary statistics.</td> <td>Cannot handle more than 16 taxa at a time, and requires subsetting larger datasets</td> <td>http://www.phrapl.org/</td> <td>(Jackson et al., 2017)</td> </tr> <tr> <td>PhyML</td> <td>Phylogeny</td> <td>Phylogenetic relationships</td> <td>Maximum Likelihood inference of phylogenetic relationships. An online version is available</td> <td>Should be used on complex of species or divergent populations with little migration. Can be ran on genomic windows to detect introgression (with e.g. TWISST, Dsuite)</td> <td>http://www.atgc-montpellier.fr/phyml/binaries.php</td> <td>(Guindon et al., 2010)</td> </tr> <tr> <td>RAxML</td> <td>Phylogeny</td> <td>Network reconstruction and phylogenetic relationships</td> <td>Maximum Likelihood inference of phylogenetic relationships</td> <td>Should be used on complex of species or divergent populations with little migration</td> <td>http://sco.h-its.org/exelixis/web/software/raxml/index.html</td> <td>(Stamatakis, 2014)</td> </tr> <tr> <td>SNAPP</td> <td>Phylogeny</td> <td>Phylogenetic relationships</td> <td>Handles SNP data</td> <td>Remains slow for medium to large datasets (&gt;1,000SNPs)</td> <td>http://beast2.org/snapp/</td> <td>(Bryant et al., 2012)</td> </tr> <tr> <td>SNPhylo</td> <td>Phylogeny</td> <td>Network reconstruction and phylogenetic relationships</td> <td>Complete pipeline from SNP filtering to tree reconstruction</td> <td>Should be used on complex of species or divergent populations with little migration</td> <td>http://chibba.pgml.uga.edu/snphylo/</td> <td>(Lee et al., 2014)</td> </tr> <tr> <td>SVDQuartets</td> <td>Phylogeny</td> <td>Phylogenetic relationships</td> <td>Estimates populations/species tree from gene trees</td> <td>Remains slow for large datasets. Requires PAUP*.</td> <td>https://www.asc.ohio-state.edu/kubatko.2/software/SVDquartets/</td> <td>(Chifman and Kubatko, 2014)</td> </tr> <tr> <td>SVDQuest</td> <td>Phylogeny</td> <td>Phylogenetic relationships</td> <td>Estimates populations/species tree from gene trees</td> <td>Faster than SVDQuartets</td> <td>https://github.com/pranjalv123/SVDquest</td> <td>(Vachaspati and Warnow, 2018)</td> </tr> <tr> <td>* BEAST</td> <td>Phylogeny and species tree inference</td> <td>Divergence time estimation and phylogenetic relationships</td> <td>Outputs a species tree instead of concatenated gene tree. Allows for testing consistency between phylogenetic signals at different loci</td> <td>Slow for large datasets. Requires sequence data. Not suited for situations where gene flow/admixture is important</td> <td>http://beast2.org/</td> <td>(Heled and Drummond, 2010)</td> </tr> <tr> <td>Splitstree</td> <td>Phylogeny/Network</td> <td>Network reconstruction and phylogenetic relationships</td> <td>User friendly interface, proposes a variety of methods for networks reconstruction</td> <td>Mostly descriptive</td> <td>http://www.splitstree.org/</td> <td>(Huson and Bryant, 2006)</td> </tr> <tr> <td>diCal2</td> <td>Sequentially Markovian coalescent</td> <td>Testing any arbitrary demographic scenario</td> <td>Works with smaller, more fragmented datasets than PSMC. Handles more complex demographic models than MSMC (including admixture).</td> <td>Requires phased whole genome data and a model to be defined</td> <td>https://sourceforge.net/projects/dical2/</td> <td>(Sheehan et al., 2013)</td> </tr> <tr> <td>MSMC and MSMC-IM</td> <td>Sequentially Markovian coalescent</td> <td>Inferring change in Ne and migration rates with time between two populations</td> <td>Allows to track population size changes in time without a priori. Allows estimating variation in cross-coalescence rate between two populations</td> <td>Limited to the study of 8 diploid individuals from 2 populations at once. Requires whole genome phased data and masking regions with insufficient sequencing depth</td> <td>https://github.com/stschiff/msmc and https://github.com/wangke16/MSMC-IM</td> <td>(Schiffels and Durbin, 2014)</td> </tr> <tr> <td>PSMC</td> <td>Sequentially Markovian coalescent</td> <td>Inferring change in effective population sizes (Ne) with time using a single diploid genome</td> <td>Allows to track population size changes in time without a priori.</td> <td>Limited to one population and one diploid individual. Better used within MSMC. Requires phased whole genome data and masking regions with insufficient sequencing depth</td> <td>https://github.com/lh3/psmc</td> <td>(Li and Durbin, 2011)</td> </tr> <tr> <td>SMC++</td> <td>Sequentially Markovian coalescent</td> <td>Inferring change in Ne with time and splitting time between two populations</td> <td>Can analyze hundreds of individuals at a time and does not require phasing</td> <td>Masking regions as in MSMC. The ancestral allele is assumed to be the reference allele by default. Assumes a clean split for populations divergence. Future versions should allow gene flow inference.</td> <td>https://github.com/popgenmethods/smcpp</td> <td>(Terhorst et al., 2016)</td> </tr> <tr> <td>TWISST</td> <td>Topology weighting</td> <td>Chromosome painting, clustering and branching between populations</td> <td>Retrieves the most likely coalescence pattern between several taxa along the genome. Can be seen as an extension of the ABBA/BABA test</td> <td>Needs a priori grouping of individuals into taxa. Requires at least 4 taxa. Impractical for more than 6 taxa. Windows size must include enough SNPs to retrieve the correct topology but at the risk that regions with different histories are included</td> <td>https://github.com/simonhmartin/twisst</td> <td>(Martin and Van Belleghem, 2016)</td> </tr> </tbody> </table> <div class="caption"> Table 3: Summary of methods dedicated to reconstruct demographic history </div> <table> <thead> <tr> <th>Software</th> <th>Class of method</th> <th>Purpose</th> <th>Specifics</th> <th>Issues and warnings</th> <th>Link</th> <th>Reference</th> </tr> </thead> <tbody> <tr> <td>abc/abcrf</td> <td>ABC</td> <td>Performs all steps for model-checking and parameters estimation for ABC analyses. abcrf includes random forest methods (a type of supervised machine-learning)</td> <td>Informative vignette, allows graphical representation, complete and robust</td> <td>Does not perform coalescent simulations (but can be used in combination with coala)</td> <td>https://cran.r-project.org/web/packages/abc/index.html ; https://cran.r-project.org/web/packages/abcrf/index.html</td> <td>(Csill√©ry et al., 2012; Raynal et al., 2019)</td> </tr> <tr> <td>ABCToolbox</td> <td>ABC</td> <td>Complete ABC analysis, from simulations to model checking and parameters estimation</td> <td>Modular, facilitates the computation of summary statistics</td> <td>NA</td> <td>https://bitbucket.org/wegmannlab/abctoolbox/wiki/Home</td> <td>(Wegmann et al., 2010)</td> </tr> <tr> <td>DIYABC</td> <td>ABC</td> <td>Complete ABC analysis, from simulations to model checking and parameters estimation</td> <td>User-friendly. Many ways to check goodness-of-fit. Good introduction to ABC models.</td> <td>Does not model continuous gene flow.</td> <td>http://www1.montpellier.inra.fr/CBGP/diyabc/</td> <td>(Cornuet et al., 2008)</td> </tr> <tr> <td>PopSizeABC</td> <td>ABC</td> <td>Inferring change in Ne using whole-genome data</td> <td>Supposed to better assess recent events. Uses a set of summary statistics for the AFS and LD between markers. Handles multiple individuals</td> <td>Approximate bayesian approaches do not retrieve the whole information</td> <td>https://forge-dga.jouy.inra.fr/projects/popsizeabc/</td> <td>(Boistard et al., 2016)</td> </tr> <tr> <td>coala</td> <td>ABC/coalescent simulations</td> <td>Combining coalescent simulators within a single framework</td> <td>Facilitates the building of scenarios and computes summary statistics for simulations. Can be easily combined with the abc or abc-rf packages in R.</td> <td>Includes so far ms, msms and scrm</td> <td>https://cran.r-project.org/web/packages/coala/index.html</td> <td>(Staab and Metzler, 2016)</td> </tr> <tr> <td>FacSexCoalescent</td> <td>coalescent simulations</td> <td>Simulate demographic scenarios for asexual/facultatively sexual species</td> <td>Can handle varying levels of sexual reproduction, inbreeding, selfing and cloning.</td> <td>Does not handle population size changes nor selection yet.</td> <td>https://github.com/MattHartfield/FacSexCoalescent ; https://github.com/MattHartfield/FacSexCoalescentR</td> <td>(Hartfield et al., 2016)</td> </tr> <tr> <td>fastsimcoal2</td> <td>coalescent simulations</td> <td>Building any arbitrary scenario using a coalescent framework</td> <td>Any arbitrary scenario can be implemented. Handles SNP, microsatellites and sequence data.</td> <td>Does not handle selection. Slower than ms with no recombination, much faster with recombination (see manual)</td> <td>http://cmpg.unibe.ch/software/fastsimcoal2/</td> <td>(Excoffier and Foll, 2011)</td> </tr> <tr> <td>ms, msms, msABC</td> <td>coalescent simulations</td> <td>Building any arbitrary scenario using a coalescent framework</td> <td>Any arbitrary scenario can be implemented. Handles SNP, microsatellites and sequence data. msms can include selection in the model.</td> <td>Syntax can be difficult to handle for new users compared to, e.g., fastsimcoal2 (but see coala)</td> <td>http://www.bio.lmu.de/~pavlidis/home/?Software:msABC</td> <td>(Hudson, 2002; Ewing and Hermisson, 2010; Pavlidis et al., 2010)</td> </tr> <tr> <td>msms</td> <td>coalescent simulations</td> <td>Simulate demographic scenarios including selection</td> <td>Flexible, syntax similar to ms, handles arbitratily complex models. Can be used in an ABC framework to include selection as a parameter to be estimated</td> <td>Syntax can be difficult to handle for the naive user (but see coala)</td> <td>http://www.mabs.at/ewing/msms/index.shtml</td> <td>(Ewing and Hermisson, 2010)</td> </tr> <tr> <td>scrm</td> <td>coalescent simulations</td> <td>Fast simulation of chromosome-scale sequences</td> <td>Syntax similar to ms, handles any arbitrary scenario</td> <td>Does not handle gene conversion and fixed number of segregating sites (unlike ms)</td> <td>https://scrm.github.io/</td> <td>(Staab et al., 2015)</td> </tr> <tr> <td>SPLATCHE3</td> <td>coalescent simulations</td> <td>Simulating demographic scenarios in their spatial context</td> <td>Coalescent simulator for genetic data, forward-in-time for demography in space. Spatially explicit.</td> <td>Simulations can be slow (&gt;1 hour) for large datasets (&gt;100,000 SNPs) over more than 1,000 generations. Does not incorporate selection</td> <td>http://www.splatche.com/splatche3</td> <td>(Currat et al., 2019)</td> </tr> <tr> <td>Coalescence</td> <td>discoal</td> <td>Simulate selective sweeps under arbitrary demographic scenarios</td> <td>Relatively fast for short genomic fragments. Designed to simulate ‚Äúhard‚Äù and ‚Äúsoft‚Äù sweeps.</td> <td>Mostly used with diploS/HIC. Other simulators such as msms may be more suited for some scenarios.</td> <td>https://github.com/kr-colab/discoal</td> <td>(Kern and Schrider, 2016)</td> </tr> <tr> <td>QuantiNemo2</td> <td>Forward-in-time simulations</td> <td>Simulating demographic and selection scenarios in their spatial context</td> <td>Comprehensive simulator.Designed for the study of selection in a spatially-explicit context. Simulates quantitative traits, fitness landscapes and underlying genetic variation with migration. Include both population and individual-based simulations</td> <td>Scan be slow for large/complex models</td> <td>https://www2.unil.ch/popgen/softwares/quantinemo/</td> <td>(Currat et al., 2019)(Neuenschwander et al., 2019)</td> </tr> <tr> <td>SliM3</td> <td>Forward-in-time simulations</td> <td>Simulating genomic sequences with intrinsic and extrinsic factors</td> <td>One of the most comprehensive simulators. Can simulate genetic data in their spatio-temporal context, the effects of selection at linked sites, coding and non-coding variation, inbreeding and selfing. Supports tree-sequence recording for faster simulations. Large community.</td> <td>Slow for large genomic regions/large populations</td> <td>https://messerlab.org/slim/</td> <td>(Haller and Messer, 2019)</td> </tr> <tr> <td>diploS/HIC</td> <td>Supervised Machine Learning</td> <td>Detecting selective sweeps</td> <td>Classifies genomic windows as neutral, selected, or impacted by selection at linked sites. Also distinguishes between selection on standing and de novo variation. Uses a set of summary statistics describing frequency spectrum and LD, does not require phasing. Good tutorial explaining the pipeline.</td> <td>Good performance depends on the parameters used to simulate sweeps (window size, selective coefficient, demography). Requires some trial and error for new model species. Interpretation of ‚Äúsoft‚Äù and ‚Äúhard‚Äù sweeps remains discussed.</td> <td>https://github.com/kr-colab/diploSHIC</td> <td>(Schrider and Kern, 2016; Kern and Schrider, 2018)</td> </tr> <tr> <td>evoNet</td> <td>Supervised Machine Learning</td> <td>Detecting selective sweeps, balancing selection, and estimate demographic history</td> <td>Uses deep-learning algorithms to classify genomic regions as selected or neutral, and estimate effective population sizes. Flexible (any number of summary statistics can be provided by the investigator).</td> <td>Requires summary statistics as an input. Difficult for a na√Øve user.</td> <td>https://sourceforge.net/projects/evonet/?source=typ_redirect</td> <td>(Sheehan and Song, 2016)</td> </tr> <tr> <td>FastEPRR</td> <td>Supervised Machine Learning</td> <td>Estimating effective recombination rates</td> <td>Uses regression to estimate effective recombination rates from SNP alignments. Can use the VCF format. No clear bias due to phasing errors observed. Can incorporate demographic history (using ms command line).</td> <td>Requires phased data.</td> <td>https://www.picb.ac.cn/evolgen/softwares/index.html</td> <td>(Gao et al., 2016)</td> </tr> <tr> <td>FILET</td> <td>Supervised Machine Learning</td> <td>Detecting introgression</td> <td>Uses Extra Trees classifiers and dedicated summary statistics to classify genomic windows as being introgressed or not. Identifies the direction of introgression.</td> <td>Targets pulse of introgression rather than continuous gene flow, but can detect the latter. Requires phased data in a fasta format.</td> <td>https://github.com/kr-colab/FILET</td> <td>(Gao et al., 2016)</td> </tr> <tr> <td>genomatnn</td> <td>Supervised Machine Learning</td> <td>Detecting adaptive introgression</td> <td>Uses convolutional neural networks to identify adaptive introgression. Trained using the tree-sequence records obtained from SliM3. Can handle VCF files and unphased data.</td> <td>Strong computational bottleneck with SliM simulations.</td> <td>https://github.com/grahamgower/genomatnn</td> <td>(Gower et al., 2020)</td> </tr> <tr> <td>ImaGene</td> <td>Supervised Machine Learning</td> <td>Detecting selective sweeps</td> <td>Uses convolutional neural networks to classify genomic windows in bins of distinct selection coefficients. Directly uses the image of the alignment, avoiding compression (i.e. using summary statistics)</td> <td>Can be slow for large datasets.</td> <td>https://github.com/mfumagalli/ImaGene</td> <td>(Torada et al., 2019)</td> </tr> <tr> <td>RELERNN</td> <td>Supervised Machine Learning</td> <td>Estimating recombination rates</td> <td>Uses recurrent neural networks to estimate recombination rates from SNP alignments. Handles unphased and pooled data. Uses msprime (Python implementation of ms) to generate simulations upon which the algorithm is trained. Can incorporate known demographic history provided by the user.</td> <td>Can be computationnally intensive for large effective population sizes. Accuracy on pooled data is modest for low depth of coverage. Absolute estimates of recombination rates depends on the accuracy of the mutation rate used for simulations.</td> <td>https://github.com/kr-colab/ReLERNN</td> <td>(Adrion, Galloway, et al., 2020)</td> </tr> <tr> <td>SWiFr</td> <td>Supervised Machine Learning</td> <td>Detecting selective sweeps</td> <td>Uses averaged one-dependence estimator to classify genomic regions as selected or neutral. Flexible in terms of which summary statistics are used. Can incorporate demographic history.</td> <td>Requires summary statistics as an input. Only distinguishes between selective sweeps and neutral regions.</td> <td>https://github.com/ramachandran-lab/SWIFr/blob/master/README.md</td> <td>(Sugden et al., 2018)</td> </tr> </tbody> </table> <div class="caption"> Table 4: Summary of simulators and simulation-based methods for selection and demographic inference. </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> ¬© Copyright 2025 Yann X.C. Bourgeois. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. Photos from <a href="https://unsplash.com" target="_blank" rel="external nofollow noopener">Unsplash</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js?a0db7e5d5c70cc3252b3138b0c91dcaf" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script defer src="https://cdn.jsdelivr.net/npm/bootstrap-table@1.22.4/dist/bootstrap-table.min.js" integrity="sha256-4rppopQE9POKfukn2kEvhJ9Um25Cf6+IDVkARD0xh78=" crossorigin="anonymous"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?e0514a05c5c95ac1a93a8dfd5249b92e"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script> <script src="/assets/js/mathjax-setup.js?70d799092f862ad98c7876aa47712e20"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script defer src="/assets/js/progress-bar.js?2f30e0e6801ea8f5036fa66e1ab0a71a" type="text/javascript"></script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>
    addBackToTop();
  </script> <script type="module" src="/assets/js/search/ninja-keys.min.js?a3446f084dcaecc5f75aa1757d087dcf"></script> <ninja-keys hidebreadcrumbs noautoloadmdicons placeholder="Type to start searching"></ninja-keys> <script src="/assets/js/search-setup.js?6c304f7b1992d4b60f7a07956e52f04a"></script> <script src="/assets/js/search-data.js"></script> <script src="/assets/js/shortcut-key.js?6f508d74becd347268a7f822bca7309d"></script> </body> </html>